src/utils/checkMobile.tsx

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+
+type deviceProps = 'mobile' | 'tablet' | 'desktop';
+
+const useCheckResponsiveScreen = (device: deviceProps) => {
+  const [width, setWidth] = useState(window.innerWidth);
+  const handleWindowSizeChange = () => {
+    setWidth(window.innerWidth);
+  };
+
+  useEffect(() => {
+    window.addEventListener('resize', handleWindowSizeChange);
+    return () => {
+      window.removeEventListener('resize', handleWindowSizeChange);
+    };
+  }, []);
+
+  switch (device) {
+    case 'mobile':
+      return width <= 640;
```

Comment:
Might be a good idea to make 640 (and 1024, below) into named constants. Do these correspond to breakpoints?

Reply:
Yep those are EASI breakpoints, but currently defined in scss.  Makes sense to have them exist as constants here in tsx and accessible elsewhere.

Reply:
I looked into this a bit more; there's a way to import the variables from the scss file, using [CSS module support](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/). Changes necessary:
1. Rename `src/stylesheets/_variables.scss` to `src/stylesheets/_variables.module.scss`.
2. Add the following to `src/stylesheets/_variables.module.scss`:
```
// division by 1px removes the unit
:export {
  mobile: $mobile / 1px;
  tablet: $tablet / 1px;
  desktop: $desktop / 1px;
}
```
3. In `src/utils/checkMobile.tsx`, add `import deviceBreakpoints from 'stylesheets/_variables.module.scss';` to grab the exports from the `_variables` SCSS module, then change the definitions of `mobile` and `tablet` to
```
export const mobile: number = parseInt(deviceBreakpoints.tablet, 10);
export const tablet: number = parseInt(deviceBreakpoints.desktop, 10);
```
4. Edit `craco.config.js` (at the top level of the repo) to reference `src/stylesheets/_variables.module.scss` instead of `src/stylesheets/_variables.scss`.

Up to you if you want to pursue this, or maybe just put a comment in `checkMobile.tsx` noting that the breakpoints are also defined in `src/stylesheets/_variables.scss`.

Reply:
Ah, very nice!  It's best to have a central variable for these, so I'll check it out.